### PR TITLE
gloas: add modified slot processing

### PR DESCRIPTION
**Pull request by terencechain**
This PR implements [process_slot](https://github.com/ethereum/consensus-specs/blob/master/specs/gloas/beacon-chain.md#modified-process_slot) and spec tests

### CODE REVIEW

File: beacon-chain/core/transition/transition_gloas_test.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+	availability := bytes.Repeat([]byte{0xFF}, int(cfg.SlotsPerHistoricalRoot/8))
+	st := newGloasState(t, slot, availability)
+
+	_, err := ProcessSlot(context.Background(), st)
+	require.NoError(t, err)
+
+	post := st.ToProto().(*ethpb.BeaconStateGloas)
+	require.Equal(t, byte(0xFF)&^bitMask, post.ExecutionPayloadAvailability[byteIdx])
+}
+
```
Comment (satushh) [inline beacon-chain/core/transition/transition_gloas_test.go:34]: We could add another test for edge case slot value. 
```suggestion
func TestProcessSlot_GloasWraparound(t *testing.T) {
	cfg := params.BeaconConfig()
	// Set slot to the last slot before wraparound
	slot := cfg.SlotsPerHistoricalRoot - 1
	nextIdx := uint64((slot + 1) % cfg.SlotsPerHistoricalRoot)
	require.Equal(t, uint64(0), nextIdx, "expected wraparound to index 0")
	byteIdx := nextIdx / 8
	bitMask := byte(1 << (nextIdx % 8))
	availability := bytes.Repeat([]byte{0xFF}, int(cfg.SlotsPerHistoricalRoot/8))
	st := newGloasState(t, slot, availability)

	_, err := ProcessSlot(context.Background(), st)
	require.NoError(t, err)

	post := st.ToProto().(*ethpb.BeaconStateGloas)
	require.Equal(t, byte(0xFF)&^bitMask, post.ExecutionPayloadAvailability[byteIdx],"bit 0 should be cleared after wraparound")
}
```

**Review by james-prysm — DISMISSED** [review]
PR seems good, you can ping for reapproval based on accepting satyajit's comment or not

File: beacon-chain/core/transition/transition_gloas_test.go (hunk 1 above)
Comment (terencechain) [reply beacon-chain/core/transition/transition_gloas_test.go:34]: thanks for the suggestion, I added in b150c8f3fd

**Review by satushh — DISMISSED** [review]

**Review by james-prysm — DISMISSED** [review]

**Review by potuz — APPROVED** [review]
The field `ExecutionPayloadAvailability`  is 1024 bytes long and it's not a shared reference, not sure about this one, it requires hashing on every slot for 32 roots.

**Review by satushh — APPROVED** [review]

**Review by james-prysm — APPROVED** [review]

**Comment by terencechain** [comment]
> The field `ExecutionPayloadAvailability` is 1024 bytes long and it's not a shared reference, not sure about this one, it requires hashing on every slot for 32 roots.

my intuition is this is cheap and not worth adding shared ref over but happy to change it if anyone wants me to do it over
